### PR TITLE
Build Fixes & Tweaks

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Description
Running the following command,
`./Utils/trigger_content_build_with_time_to_live.sh content_branch_name <CircleCiToken> 420 contrib_name:contrib_branch_name Packs/New_Pack/`
is supposed to trigger a build that will take the contribution and will run a regular content build using the contributor's code/additions. Currently, the build does not actually use the contributor's code/additions for testing on server instances. This PR implements proper handling in our build for this usecase. See this [build](https://app.circleci.com/pipelines/github/demisto/content/19257/workflows/f7a00841-2b11-4a62-a002-7a1410eb84fb) (which was triggered by `./Utils/trigger_content_build_with_time_to_live.sh code42_contrib <circle-token-redacted> 420 code42:download-file Packs/Code42/`) for an example where the contributor's code was not properly fetched for testing on server instances.

## Minimum version of Demisto
- [x] 4.5.0
- [x] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

**Comment by avidan-H** [comment]
I'm adding the `docs-approved` label since there is no documentation to review.

File: .circleci/config.yml [hunk 1]
```diff
@@ -180,12 +184,13 @@ references:
       paths:
         - project
 
-  Secrets: &Secrets
+  secrets: &secrets
```
Comment (Itay4) [inline .circleci/config.yml:187]: why did you lower case it? i think it is better shown in the step as `Secrets`

File: .circleci/config.yml [hunk 2]
```diff
@@ -273,21 +284,41 @@ references:
...
     run:
       name: Update Tests step
       when: always
       command: |
         python3 ./Tests/scripts/update_conf_json.py
         cp "./Tests/conf.json" "$CIRCLE_ARTIFACTS/conf.json"
+
+  get_contribution_pack: &get_contribution_pack
+    when:
+      condition: << pipeline.parameters.contrib_branch >>
```
Comment (Itay4) [inline .circleci/config.yml:305]: should we use `CONTRIB_BRANCH` here ?

File: Tests/scripts/prepare_content_packs_for_testing.sh [hunk 3]
```diff
@@ -37,10 +37,6 @@ echo "Updating modified content packs in the bucket ..."
 
 if [ ! -n "${NIGHTLY}" ]; then
     CONTENT_PACKS_TO_INSTALL_FILE="./Tests/content_packs_to_install.txt"
-
-  if [ -n "${CONTRIB_BRANCH}" ]; then
```
Comment (yaakovi) [inline Tests/scripts/prepare_content_packs_for_testing.sh:41]: why did you remove this?
it was added because the `Collect tests and packs to install` did not recognize it as a changed pack (as this might be a new pack and not part of the repo)

File: .circleci/config.yml (hunk 1 above)
Comment (avidan-H) [reply .circleci/config.yml:187]: image
In the step itself, it is capitalized - I only changed the alias to be lowercase in keeping with the format of all of the other aliases.

File: .circleci/config.yml (hunk 2 above)
Comment (avidan-H) [reply .circleci/config.yml:305]: We set this env variable in the `environment` alias step - if a job doesn't use this alias for setting the environment, then the variable will not be set. I think it's preferable to use the pipeline parameter since that should always be present.

File: Tests/scripts/prepare_content_packs_for_testing.sh (hunk 3 above)
Comment (avidan-H) [reply Tests/scripts/prepare_content_packs_for_testing.sh:41]: Cause this would mistakenly add the contributor's pack name to the `content_packs_to_install.txt` file even when the contributor's pack wasn't pulled (which was the case in our build). The proper way to do this is to pull in the contributor's changes and have those changes detected in the `collect_test_list_and_content_packs` step the same way we do for all of our content.
